pgmpy/inference/base.py

```diff
@@ -80,3 +81,28 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+        elif isinstance(model, DynamicBayesianNetwork):
+            # This method constructs a temporary Bayesian Model so as to create the
+            # initial potential for inference.
+            # This will also declare the following two parameters
+            # 1) interface nodes:- These are the nodes whose children are in the 
+            # first timeslice.
+            # 2) one_and_half_model:- This is the bayesian model which consists of the
+            # interface nodes along with the first time slice nodes.
+            # Both of the above parameters will be required by the junction tree algorithm
+            # too.
+            start_model = BayesianModel(model.edges())
+            start_model.add_cpds(*model.cpds)
+            self.start_bayesian_model = start_model
+            # interface nodes extraction for inference
+            self.interface_nodes = [(edge[0][0], 0) for edge in model.get_inter_edges()]
+            slice_1_interface_nodes = [(edge[0][0], 1) for edge in model.get_inter_edges()]
```

Comment:
This can be done in a single line with a single loop: `self.interface_nodes, slice_1_interface_nodes = zip(*[((edge[0][0],  0), (edge[0][0], 1)) for edge in model.get_inter_edges()])`